### PR TITLE
Issues/434/hide rng

### LINKED ISSUE #434: Pull random generator code out of sorcha.py
Enable env variable for reproducible tests.

### PR DESCRIPTION
**Pull request by maxwest-uw**
Fixes #434 .

creates a dataclass for the command line arguments, which has an `_rng` component. the numpy rng _cannot_ be set through the the dict or the .ini config file, it must be set manually after creating a `sorchaArguments` object, like

```python
args = sorchaArguments(cmd_args_dict)
args._rng = np.random.default_rng(YOUR_TEST_SEED)
runLSSTPostProcessing(args)
```

any users/grad students would have to dig deep into the code and ignore multiple warnings to mess this up.

the arguments dataclass doesn't drastically change the api of the code either, as you can still pass a `dict` of arguments to `runLSSTPostProcessing` and it will create a `sorchaArguments` object for you.

Describe your changes.

## Review Checklist for Source Code Changes

- [x] Does pip install still work?
- [x] Have you written a unit test for any new functions?
- [x] Do all the units tests run successfully?
- [x] Does Sorcha run successfully on a test set of input files/databases?
- [x] Have you used black on the files you have updated to confirm python programming style guide enforcement?

### CODE REVIEW

File: src/sorcha/utilities/sorchaArguments.py
Comment (drewoldag) [inline src/sorcha/utilities/sorchaArguments.py:45]: 💯

File: src/sorcha/utilities/sorchaArguments.py
Comment (drewoldag) [inline src/sorcha/utilities/sorchaArguments.py:30]: I don't think we should change it, but this doesn't seem like an ideal variable name.

**Review by drewoldag — APPROVED** [review]
This looks great to me. Thanks a bunch for putting this together. I'm a huge fan of dataclasses for this purpose, and I really appreciate you putting in the work here :)

**Comment by maxwest-uw** [comment]
going to wait on the CI to be fixed before merging this 😗

**Comment by drewoldag** [comment]
After some fixes yesterday the CI is running again. I also took a moment to resolve a couple of merge conflicts that cropped up in `sorcha.py`. @maxwest-uw if you have a chance to take a I look I would appreciate it, and feel free to push the squash-n-merge button. No rush, of course :)

**Pull request merged**
